libs/utils/src/properties.c

```diff
@@ -17,55 +17,87 @@
  * under the License.
  */
 
+#include "properties.h"
+#include "celix_properties.h"
+#include "celix_properties_private.h"
+
+#include <assert.h>
+#include <errno.h>
+#include <math.h>
+#include <stdbool.h>
 #include <stdio.h>
-#include <string.h>
 #include <stdlib.h>
-#include <ctype.h>
-#include <stdbool.h>
+#include <string.h>
 
-#include "properties.h"
 #include "celix_build_assert.h"
-#include "celix_properties.h"
+#include "celix_err.h"
+#include "celix_string_hash_map.h"
 #include "celix_utils.h"
-#include "utils.h"
-#include "hash_map_private.h"
-#include <errno.h>
-#include "hash_map.h"
+#include "celix_stdlib_cleanup.h"
+#include "celix_convert_utils.h"
 
-#define MALLOC_BLOCK_SIZE        5
+#define CELIX_SHORT_PROPERTIES_OPTIMIZATION_STRING_BUFFER_SIZE 512
```

Comment:
Short properties optimization is a good idea. I am not sure whether the size should be set according to max values.  If this is enough for 80% usages, then I consider it a good choice. Given that celix_properties are used extensively and for various purposes other than service metadata, 128/256 may seems less scary.

Reply:
I will make this configurable (as is done with `celix_err_buffer_size`) and set the default to 128. 

I also created a issue to #684 so that users can get some statistics about the service framework and use that to determine (if needed) the ideal properties optimization size for their use cases.